src/Framework/FieldsAPI/Checkbox.php

```diff
@@ -2,48 +2,112 @@
 
 namespace Give\Framework\FieldsAPI;
 
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
+
 /**
- * @since 2.12.0
+ * A single checkbox input. If supported, an indeterminate value is represented as a null value, as opposed to true or
+ * false value.
+ *
+ * @unreleased
  */
 class Checkbox extends Field
 {
-
-    use Concerns\HasEmailTag;
     use Concerns\HasHelpText;
     use Concerns\HasLabel;
-    use Concerns\HasOptions;
-    use Concerns\HasPlaceholder;
     use Concerns\ShowInReceipt;
     use Concerns\StoreAsMeta;
 
     const TYPE = 'checkbox';
 
     /**
-     * Helper for marking the checkbox as checked by default
+     * @var bool|null
+     */
+    protected $defaultValue = false;
+
+    /**
+     * @var bool
+     */
+    protected $supportIndeterminateValue = false;
+
+    /**
+     * @unreleased
      *
-     * @since 2.12.0
+     * @param bool|null $defaultValue
      *
-     * @param bool|callable $isChecked
+     * @return void
+     */
+    public function defaultValue($defaultValue = null)
+    {
+        if ($defaultValue === null && !$this->supportIndeterminateValue) {
+            throw new InvalidArgumentException(
+                'Checkbox field does not support intermediate value'
+            );
+        }
+
+        if (!is_bool($defaultValue) && $defaultValue !== null) {
+            throw new InvalidArgumentException(
+                'Checkbox field only supports boolean or null value'
+            );
+        }
+
+        $this->defaultValue = $defaultValue;
+    }
+
+    /**
+     * Sets the checkbox as checked by default
      *
-     * @return $this
+     * @unreleased
      */
-    public function checked($isChecked = true)
+    public function checked(bool $checked = true): self
     {
-        $default = is_callable($isChecked) ? $isChecked() : $isChecked;
-        $this->defaultValue((bool)$default);
+        $this->defaultValue($checked);
 
         return $this;
     }
 
     /**
-     * Returns whether the checkbox is checked by default
+     * Sets the checkbox as indeterminate by default
      *
-     * @since 2.12.0
-     *
-     * @return bool
+     * @unreleased
+     */
+    public function indeterminate(): self
+    {
+        $this->defaultValue(null);
+
+        return $this;
+    }
```

Comment:
indeterminate is a special checkbox's state, not it's value.

Reply:
Good catch. This was an interesting lesson for me in exactly _what_ an indeterminate state is! I've removed this whole concept from the field.